assets/js/src/marketing_optin_block/block.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/react-in-jsx-scope */
+/* eslint-disable import/no-unresolved */
+
+/**
+ * External dependencies
+ */
+import { CheckboxControl } from '@woocommerce/blocks-checkout';
+import { useState, useEffect } from '@wordpress/element';
```

Comment:
I am not sure why are you using `useState` and `useEffect` from `'@wordpress/element'`. In other places, we use the React functions. What is the benefit?

Reply:
We follow the WordPress/Gutenberg standard of using core packages instead of directly referencing React, `@wordpress/element` is a light superset of React which allows the Gutenberg team to control which version of React is served (to avoid version conflict), offers extra functions, and allows you down the line to replace the react implementation if needed (like Jetpack using Preact).

I understand that your team uses React in your custom screens, but given this code is going to be served and consumed in Gutenberg, I feel it might less dangerous to just use `@wordpress/element` to avoid any version conflict.